server/test/integration/vitalsTest/handleHeartbeatTest.js

```diff
@@ -56,6 +61,54 @@ async function lowBatteryHeartbeat(coreId) {
 }
 
 describe('vitals.js integration tests: handleHeartbeat', () => {
+  describe('for a request with an incorrect API key', () => {
+    beforeEach(async () => {
+      await db.clearTables()
+
+      const client = await factories.clientDBFactory(db)
+      await locationDBFactory(db, {
+        locationid: testLocation1Id,
+        radarCoreId: radar_coreID,
+        clientId: client.id,
+      })
+
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+      sandbox.spy(helpers, 'logSentry')
+      sandbox.stub(db, 'logSensorsVital')
+    })
+
+    afterEach(async () => {
+      await db.clearTables()
+      sandbox.restore()
+    })
+
+    it('should return 200 for the valid request', async () => {
+      const badRequest = {
+        coreid: radar_coreID,
+        data: `{"doorMissedMsg": 0, "doorLowBatt": true, "doorLastMessage": 1000, "resetReason": "NONE", "states":[]}`,
+        api_key: badpassword,
+      }
+      const response = await chai.request(server).post('/api/heartbeat').send(badRequest)
+      expect(response).to.have.status(200)
+      it('should log the error because the api key does not match', () => {
+        expect(helpers.log).to.have.been.calledWith(`Access not allowed to: incorrect key ${badpassword}`)
+      })
```

Comment:
This is a separate test, so would be better at the same level as the other `it` functions.

Whenever an `expect` fails, then it will return out of the `it` function that it's in. Therefore, if the `expect(response).to.have.status(200)` on line 93 fails, then this test will never run. Whereas, we would want this code to run and know whether the error was logged regardless of whether the right status code was being returned or not.

Reply:
ohhhh okay that makes sense. thanks! i'll move it out of this level!